### DISCUSSION TITLE
larger sample files

### DISCUSSION THREAD

**Discussion by sbenphia** [CMSgov/price-transparency-guide#493] [General]
I tried searching a bit and couldn't find if this topic has come up previously -- are there any larger sample files available that represent what these published files may contain?  If not publicly available, does CMS have a script they used to generate larger sample files (perhaps that used a public provider and hcpcs code list to be realistic) that could be made available.

On that note - if anyone is posting actual large files a little early (one that passed the validator), it would be great to know and this forum would be a good place to post about that successful achievement so others can take a look as they get ready to post.

**Comment by shaselton-usds** [comment]
To test the validator's speed on local machines, we took one of the sample INN files and repeated the in_network object up to a million times to generate a file in the GB size. 

Github doesn't allow for files to be uploaded in a typical repo that would match some of the size estimates that we've seen given here.

What have you been seeing when generating initial files on your end? Have you been running into validation issues with the validator?

**Reply by sbenphia** [reply]
We haven't gotten a machine readable file yet (as an employer) to validate, but thought it would be nice if there is a file that matches the size estimates for testing.  CMS.data.gov hosts a lot of larger datasets, perhaps put one up there - even better, generate a CMS machine readable file from the allowed amount data files that CMS already publishes (in a different format) each year.   No harm in CMS being an owner "and a member" of this club (I get that CMS isn't required to produce an MRF)

**Comment by sbenphia** [comment]
Unfortunately, we do not yet have a Machine Readable File to test.  As a self-funded employer (and consultant for other self-funded employers)  the networks/TIC vendors have assured us MRFs will be available, but we haven't received anything to validate yet, but wanted to make sure that 1) the validator would work for us to check the files, and 2) start to figure out how to read the files in some sort of streaming fashion similar to what the validator uses (especially if they are going to be "terabytes" as indicated by the networks).

The small files posted can be read in a browser with javascript, but the larger files will likely require some additional work to deal with the memory issues large files create (lots of ways to do that)

on a related note - there are a lot of large machine readable files and apis on cms.data.gov - including medicare payment data files.  What if you posted a file (generated from the already posted Medicare payment data) in MRF format on cms.data.gov - Here is where that data is - it seems like you have all the data points you would need to convert this into an "allowed-amount" MRF file with real data (instead of the million repeat file).

https://www.cms.gov/Research-Statistics-Data-and-Systems/Statistics-Trends-and-Reports/Medicare-Provider-Charge-Data/Physician-and-Other-Supplier

________________________________
